core/arch/arm/mm/sp_mem.c

```diff
@@ -46,6 +47,33 @@ struct sp_mem *sp_mem_new(void)
 	return smem;
 }
 
+bool sp_mem_is_shared(struct sp_mem_map_region *new_reg)
+{
+	struct sp_mem *smem = NULL;
+
+	SLIST_FOREACH(smem, &mem_shares, link) {
```

Comment:
Do we need to hold `sp_mem_lock` when accessing `mem_shares`?

Reply:
Done